crates/dies-world/src/filter/kalman.rs

```diff
@@ -0,0 +1,167 @@
+use nalgebra as na;
+use na::{DimName, OMatrix};
+use na::{DefaultAllocator};
+use na::allocator::Allocator;
+use na::OVector;
+use crate::filter::matrix_gen::MatrixCreator;
+
+// OS: Observation Space, SS: State Space
+#[allow(non_snake_case)]
+pub struct Kalman<OS,SS>
+where
+    OS: DimName,
+    SS: DimName,
+    DefaultAllocator: Allocator<f64, SS, SS> + Allocator<f64, OS, SS>
```

Comment:
We can get rid of these trait constraints if we use SMatrix/SVector instead of the owned versions. I don't think there's a reason we need the owned types here. What do you think?

Reply:
I had problems with SMatrix and generic types, for example, if I define `R: OMatrix<f64, OS, OS>` it gives **Type provided when a constant was expected [E0747]** about OS. I tried several ways and they all didn't work, I don't know if the problem is solvable

Reply:
To use SMatrix you need to declare the type parameter as const:
...<const OS: usize, ...>

Reply:
https://doc.rust-lang.org/reference/items/generics.html#const-generics